api/dp-openapi.yaml

```diff
@@ -1057,6 +1127,44 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/pages/{pagename}/wordcheck:
+    put:
+      tags:
+      - project
+      description: Report wordcheck reults
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/state_query'
```

Comment:
I still don't understand why the caller needs to pass in the project or page state. It either is in a state where the user can make this report -- and the API call passes -- or it isn't and the API call fails.

Reply:
I think this applies to all the proofreading stuff. How it works now is that it first checks that the project and page are in the state the caller thinks they are in and then checks that the action can be carried out in the given states. Should we change it?

Reply:
I know it currently applies to the checking/checkout endpoint and lets not change that in this PR at least but we should remove the requirements from this endpoint.

Think about it this way: why are we asking a function call to validate our idea of the object state before doing something we think we can do. Either we can do it because the object is a state that allows it, or we can't, and the API should enforce that. Any object could change state in an async system while we weren't paying attention. This has never made sense to me in our current codebase.

Reply:
That seems a good way to look at it. I'll change this endpoint.